### PR TITLE
MAT-5998: Escape html in Human readable service, remove liquid escaping

### PR DESCRIPTION
**Pull request by mcmcphillips**
## MADiE FHIR SERVICE

Jira Ticket: [MAT-5998](https://jira.cms.gov/browse/MAT-5998)
(Optional) Related Tickets:

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included.
* [x] No extemporaneous files are included (i.e Complied files or testing results).
* [x] This PR is merging into the **correct branch**.
* [x] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [x] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/main/resources/templates/Measure.liquid [hunk 1]
```diff
@@ -167,13 +167,13 @@
         {% if Measure.publisher.exists() %}
             <tr>
                 <th scope="row"><b>Publisher: </b></th>
-                <td style="padding-left: 4px;">{{Measure.publisher.escape('html')}}</td>
+                <td style="padding-left: 4px;">{{Measure.publisher}}</td>
```
Comment (RohitKandimalla) [inline src/main/resources/templates/Measure.liquid]: Are these changes in liquid template provided by MITRE team ?

File: src/main/java/gov/cms/madie/madiefhirservice/services/HumanReadableService.java [hunk 2]
```diff
@@ -24,6 +29,110 @@ public class HumanReadableService extends ResourceUtils {
...
+    measure.setPurpose(escapeStr(measure.getPurpose()));
+    measure.setUsage(escapeStr(measure.getUsage()));
+    measure.setCopyright(escapeStr(measure.getCopyright()));
+    measure.setDisclaimer(escapeStr(measure.getDisclaimer()));
+    measure.setGuidance(escapeStr(measure.getGuidance()));
+    measure.setClinicalRecommendationStatement(
+        escapeStr(measure.getClinicalRecommendationStatement()));
+  }
+
+  public void escapeSupplementalProperties(org.hl7.fhir.r5.model.Measure measure) {
```
Comment (RohitKandimalla) [inline src/main/java/gov/cms/madie/madiefhirservice/services/HumanReadableService.java]: Does these methods have to be public ?

File: src/main/java/gov/cms/madie/madiefhirservice/services/HumanReadableService.java [hunk 3]
```diff
@@ -24,6 +29,110 @@ public class HumanReadableService extends ResourceUtils {
...
   private final LiquidEngine liquidEngine;
 
+  public String escapeStr(String val) {
+    if (val != null && !val.isEmpty()) {
+      return htmlEscape(val);
+    }
+    return val;
+  }
+
+  public void ecapeTopLevelProperties(org.hl7.fhir.r5.model.Measure measure) {
```
Comment (RohitKandimalla) [inline src/main/java/gov/cms/madie/madiefhirservice/services/HumanReadableService.java]:
```suggestion
  public void escapeTopLevelProperties(org.hl7.fhir.r5.model.Measure measure) {
```

File: src/main/java/gov/cms/madie/madiefhirservice/services/HumanReadableService.java (hunk 2 above)
Comment (RohitKandimalla) [reply src/main/java/gov/cms/madie/madiefhirservice/services/HumanReadableService.java]: Also In general it is considered to be a bad practice to update the original object.

File: src/main/resources/templates/Measure.liquid (hunk 1 above)
Comment (mcmcphillips) [reply src/main/resources/templates/Measure.liquid]: No they were our additions to the template

File: src/main/java/gov/cms/madie/madiefhirservice/services/HumanReadableService.java (hunk 2 above)
Comment (mcmcphillips) [reply src/main/java/gov/cms/madie/madiefhirservice/services/HumanReadableService.java]: I can change them to private.
In this case do you think we should make multiple r5 objects?

File: src/main/resources/templates/Measure.liquid (hunk 1 above)
Comment (RohitKandimalla) [reply src/main/resources/templates/Measure.liquid]: So the problem with our additions, is that when a new template is provided by MITRE, we will be overwriting what we have, and at that point, We will loose all these updates.

Comment (RohitKandimalla) [reply src/main/resources/templates/Measure.liquid]: I think if Liquid template is already handling escaping of html tags, and if thats not working, we should report to MITRE or create a new PR on their repo as well.

**Review by RohitKandimalla — APPROVED** [review]
